clang/lib/Sema/SemaStmtAsm.cpp

```diff
@@ -664,11 +664,22 @@ StmtResult Sema::ActOnGCCAsmStmt(SourceLocation AsmLoc, bool IsSimple,
       SmallerValueMentioned |= OutSize < InSize;
     }
 
+    // If the input is an integer register while the output is floating point,
+    // there is no way they can work together.
+    bool FPBoundToInt = false;
+    if (InputDomain != AD_FP && OutputDomain == AD_FP) {
+      FPBoundToInt = true;
+    }
+    if (InputDomain == AD_FP && OutputDomain != AD_FP) {
+      FPBoundToInt = true;
+    }
```

Comment:
Can rewrite this to one expression to initialize FPBoundToInt. Also not sure "Bound" is the right terminology to use

Reply:
I think this might be clearer than expressions like `(InputDomain == AD_FP) ^ (OutputDomain == AD_FP)`. By the way the previous parts uses "Tied", so I think I'd change it this way.

Reply:
This should be an initialization expression

Reply:
Done.